script/tool/lib/src/publish_plugin_command.dart

```diff
@@ -102,60 +119,55 @@ class PublishPluginCommand extends PluginCommand {
     }
     _print('Local repo is ready!');
 
-    await _publish();
+    final Directory packageDir = _checkPackageDir(package);
+    await _publishPlugin(packageDir: packageDir);
+    await _tagRelease(packageDir: packageDir, remote: remote, remoteUrl: remoteUrl, shouldPushTag: shouldPushTag);
+    await _finishSuccesfully();
+  }
+
+  Future<void> _publishPlugin({@required Directory packageDir}) async {
+    await _checkGitStatus(packageDir);
+    await _publish(packageDir);
     _print('Package published!');
+  }
+
+  Future<void> _tagRelease({@required Directory packageDir, @required String remote, @required String remoteUrl, @required bool shouldPushTag}) async {
     if (!argResults[_tagReleaseOption]) {
-      return await _finishSuccesfully();
+      return;
     }
-
     _print('Tagging release...');
-    final String tag = _getTag();
-    await processRunner.runAndExitOnError('git', <String>['tag', tag],
-        workingDir: _packageDir);
+    final String tag = _getTag(packageDir);
+    await processRunner.runAndExitOnError('git', <String>['tag', tag], workingDir: packageDir);
     if (!shouldPushTag) {
-      return await _finishSuccesfully();
+      return;
     }
 
     _print('Pushing tag to $remote...');
     await _pushTagToRemote(remote: remote, tag: tag, remoteUrl: remoteUrl);
-    await _finishSuccesfully();
   }
 
   Future<void> _finishSuccesfully() async {
     await _stdinSubscription.cancel();
     _print('Done!');
   }
 
-  Directory _checkPackageDir() {
-    final String package = argResults[_packageOption];
-    if (package == null) {
-      _print(
-          'Must specify a package to publish. See `plugin_tools help publish-plugin`.');
+  Directory _checkPackageDir(String package) {
```

Comment:
Can we rename this to something like _getPackageDir (and document that it throws if the dir doesn't exist)? I would expect a `check` function to either return a bool, or return nothing.